ZprimeMuMuPatMiniAodNewMC.C

```diff
@@ -1169,40 +1248,46 @@ void ZprimeMuMuPatMiniAodNewMC::PlotRecoInfo(float CosmicMuonRejec, float vertex
   */
   h2_CSSmearedMassBinned_->Fill(m_vtxMassSmearedMu,        (priEtaBin*3)+0,newweight);
   h2_CSMassBinned_       ->Fill(m_vtxMassMu,               (priEtaBin*3)+0,newweight);
-  h2_CSMassUpBinned_     ->Fill(m_vtxMassMu*(1+m_scaleUnc),(priEtaBin*3)+0,newweight);
-  h2_CSMassDownBinned_   ->Fill(m_vtxMassMu*(1-m_scaleUnc),(priEtaBin*3)+0,newweight);
+  h2_CSMassMuIDBinned_   ->Fill(m_vtxMassMu,               (priEtaBin*3)+0,newweight*SF1*SF2);
+  h2_CSMassUpBinned_     ->Fill(m_vtxMassScaledMu,         (priEtaBin*3)+0,newweight);
+  h2_CSMassDownBinned_   ->Fill(m_vtxMassScaledMu          ,(priEtaBin*3)+0,newweight);
   if (secEtaBin > 0) {
     // std::cout << "secondary bin (" << secEtaBin << "*3)+0(" << (secEtaBin*3)+0 << ")" << std::endl;
     h2_CSSmearedMassBinned_->Fill(m_vtxMassSmearedMu,        (secEtaBin*3)+0,newweight);
     h2_CSMassBinned_       ->Fill(m_vtxMassMu,               (secEtaBin*3)+0,newweight);
-    h2_CSMassUpBinned_     ->Fill(m_vtxMassMu*(1+m_scaleUnc),(secEtaBin*3)+0,newweight);
-    h2_CSMassDownBinned_   ->Fill(m_vtxMassMu*(1-m_scaleUnc),(secEtaBin*3)+0,newweight);
+    h2_CSMassMuIDBinned_   ->Fill(m_vtxMassMu,               (secEtaBin*3)+0,newweight*SF1*SF2);
```

Comment:
Is this (and elsewhere) a correction on the mass? And would it make more sense to do, e.g.,:
```
    h2_CSMassMuIDBinned_   ->Fill(m_vtxMassMu*SF1*SF2,               (secEtaBin*3)+0,newweight);
```
For another part of the code I was thinking about this in the past, and wasn't sure what the effect would be, and what the "correct" way was.

Reply:
No, SF1 and SF2 are per event weights as they represent an efficiency loss at high p. So I think multiplying it onto the weight is correct here.

Reply:
OK, makes sense, thanks!